#! /bin/sh /usr/share/dpatch/dpatch-run
## 99_fix_testsuite_for_installed_env.dpatch by  <sven@imperator.oops.nntp.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix the mysql testsuite to work in the installed locations

@DPATCH@

Common subdirectories: mysql-5.5.orig/mysql-test//.svn and mysql-5.5/mysql-test//.svn
Common subdirectories: mysql-5.5.orig/mysql-test//collections and mysql-5.5/mysql-test//collections
Common subdirectories: mysql-5.5.orig/mysql-test//extra and mysql-5.5/mysql-test//extra
Common subdirectories: mysql-5.5.orig/mysql-test//include and mysql-5.5/mysql-test//include
Common subdirectories: mysql-5.5.orig/mysql-test//lib and mysql-5.5/mysql-test//lib
diff -u mysql-5.5.orig/mysql-test//mysql-test-run.pl mysql-5.5/mysql-test//mysql-test-run.pl
--- mysql-5.5.orig/mysql-test//mysql-test-run.pl	2011-03-08 23:00:17.000000000 +0100
+++ mysql-5.5/mysql-test//mysql-test-run.pl	2011-03-08 23:11:30.000000000 +0100
@@ -3114,6 +3114,10 @@
     mtr_appendfile_to_file("$sql_dir/mysql_system_tables_data.sql",
 			   $bootstrap_sql_file);
 
+    mtr_tofile($bootstrap_sql_file, "-- Debian removed the default privileges on the 'test' database\n");
+    mtr_tofile($bootstrap_sql_file, "INSERT INTO mysql.db VALUES ('%','test','','Y','Y','Y','Y','Y','Y','N','Y','Y','Y','Y','Y','Y','Y','Y','N','N','Y','Y');\n");
+    
+
     # Add test data for timezone - this is just a subset, on a real
     # system these tables will be populated either by mysql_tzinfo_to_sql
     # or by downloading the timezone table package from our website
Common subdirectories: mysql-5.5.orig/mysql-test//r and mysql-5.5/mysql-test//r
Common subdirectories: mysql-5.5.orig/mysql-test//std_data and mysql-5.5/mysql-test//std_data
Common subdirectories: mysql-5.5.orig/mysql-test//suite and mysql-5.5/mysql-test//suite
Common subdirectories: mysql-5.5.orig/mysql-test//t and mysql-5.5/mysql-test//t
